pkg/components/ldap_auth.go

```diff
@@ -0,0 +1,69 @@
+package components
+
+import (
+	"context"
+	"net/http"
+)
+
+type authValidationTransport struct {
+	Wrapped              http.RoundTripper
+	AllowedGroups        []string
```

Comment:
task: Groups are only possible set of values that we might check here. I'd say we should pick a different name for this so that it doesn't imply that this plugin can only be used to validate the names of LDAP groups.

Reply:
What might we check beyond groups? I was under the impression we would just look for groups, although they may not be named ldap groups.

Maybe AllowedValues would be a better more generic name?